src/components/screens/vacation-requests-screen.tsx

```diff
@@ -74,10 +75,31 @@ const VacationRequestsScreen = () => {
       } catch (error) {
         setError(`${strings.vacationRequestError.fetchStatusError}, ${error}`);
       }
-      setLoading(false);
     }
+    setLoading(false);
   };
 
+  const filterVacationRequests = async () => {
+    if (loading) {
+      return;
+    }
+    setToggleFilter((prevToggleFilter) => {
+      const newToggleFilter = !prevToggleFilter;
+      const newVacationRequest = originalData.filter((vacation) =>
+        newToggleFilter
+          ? vacation.endDate.getTime() <= Date.now()
+          : vacation.endDate.getTime() > Date.now()
+      );
+      setVacationRequests(newVacationRequest);
+      return newToggleFilter;
+    });
+  };
+
+  useEffect(() => {
+    if(!originalData.length) return;
+    filterVacationRequests();
+  }, [originalData]);
+
```

Comment:
Can we make this useEffect to be triggered by changes in toggleFilter and then call this filterVacationRequest (without original data and setToggleFilter)? And not to send to child components this filterVacationRequests?